src/shared/openssl-util.c

```diff
@@ -11,6 +11,15 @@
 #include "string-util.h"
 
 #if HAVE_OPENSSL
+#include <openssl/rsa.h>
+
+#if !defined(OPENSSL_NO_ENGINE) && !defined(OPENSSL_NO_DEPRECATED_3_0)
+#include <openssl/engine.h>
```

Comment:
Maybe ident those those show that they are nested:
```C
#  include <openssl/rsa.h>
#    if !defined…
#      include <openssl/engine.h>
...
```

Reply:
Fixed. Usually just try to fit in the surrounding code...